python/adjoint/objective.py

```diff
@@ -231,98 +231,52 @@ def __call__(self):
 
 
 class FourierFields(ObjectiveQuantity):
-    def __init__(self, sim, volume, component, decimation_factor=0):
+    def __init__(self, sim, volume, component, yee_grid=False, decimation_factor=0):
```

Comment:
I think we should keep the default value of `yee_grid=True`. As discussed in #1793, it's better for the user to not worry about the Yee grid. i.e. if they specify a point, they shouldn't need to make sure it's on the Yee grid (or understand that it's being *snapped* under the hood). 

To make this work we'll need to do a restriction step for the adjoint sources, of course.